lib/sierra-models/base.js

```diff
@@ -73,25 +73,19 @@ class SierraBase {
           //    add parallel to the primary
           field.parallel = { ...parallelsLookup[subfieldSixSuffix] }
           //    set value to null, so we know that parallel has been matched
-          parallelsLookup[subfieldSixSuffix] = null
+          delete parallelsLookup[subfieldSixSuffix]
         }
       }
+      return field
     }
 
-    const primariesWithAndWithoutMatches = primaryRawFields.map((field, i) => {
-      // attach relevant parallels
-      lookup(field)
-      return field
-    })
```

Comment:
Again, confusing that something called `lookup` mutates the element. Feels like a `map` using a mutating function